File: loopr-ios/Models/Token/TokenMetadata.swift
//
//  Depth.swift
//  loopr-ios
//
//  Created by kenshin on 2018/3/6.
//  Copyright © 2018年 Loopring. All rights reserved.
//

import Foundation

class TokenMetadata {
    let type: TokenType
    let status: TokenStatus
    let symbol: String
    let name: String
    let address: String
    let unit: String
    let decimals: Int
    let precision: Int
    let burnRate: BurnRate

    init(json: JSON) {
        self.type = TokenType(rawValue: json["type"].stringValue) ?? .unknown
        self.status = TokenStatus(rawValue: json["status"].stringValue) ?? .unknown
        self.symbol = json["symbol"].stringValue
        self.name = json["name"].stringValue
        self.address = json["address"].stringValue
        self.unit = json["unit"].stringValue
        self.decimals = json["decimals"].intValue
        self.precision = json["precision"].intValue
        self.burnRate = BurnRate(json: json["burnRate"])
    }
}

File: walletsdk/src/main/java/leaf/prod/walletsdk/model/token/TokenMetadata.java
/**
 * Created with IntelliJ IDEA.
 * User: kenshin wangchen@loopring.org
 * Time: 2019-03-21 2:27 PM
 * Cooperation: loopring.org 路印协议基金会
 */
package leaf.prod.walletsdk.model.token;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class TokenMetadata {

    private TokenStatus type;

    private TokenType status;

    private String symbol;

    private String name;

    private String address;

    private String unit;

    private Integer decimals;

    private Integer precision;

    private BurnRate burnRate;
}
